soap/client.go

```diff
@@ -45,7 +45,7 @@ type Client struct {
 }
 
 // RoundTrip implements the RoundTripper interface.
-func (c *Client) RoundTrip(in, out Message) error {
+func (c *Client) RoundTrip(in, out Message, soapAction string) error {
```

Comment:
Maybe give it a context as the first param instead of this extra out-of-place soapAction param? This could later be used for cancelation and whatnot...